### PR TITLE
feat(search): search chartfile name and description

### LINKED ISSUE #11: Better `helm search`
The current `helm search` only looks at the names of charts.

Could we extend to be a little bit better? Maybe search the description?

Things that _might_ be useful to have searched:

- Chart.yaml...
  - name
  - description
- manifests/
  - metadata/labels headers
  - `image:` field (which shows docker container images by name)

### PR DESCRIPTION
**Pull request by lanej**
* Fixes #11

### CODE REVIEW

File: helm/action/list.go [hunk 1]
```diff
@@ -10,13 +10,15 @@ import (
 // List lists all of the local charts.
 func List(homedir string) {
 	md := filepath.Join(homedir, WorkspaceChartPath, "*")
+
 	charts, err := filepath.Glob(md)
-	if err != nil {
```
Comment (carmstrong) [inline helm/action/list.go]: I think the paradigm in go is `err != nil`... any reason for this change?

**Comment by carmstrong** [comment]
Good stuff, @lanej. Can you fix up the commit subjects? Squashing them into one titled `feat(search): deeper search` or something like that should be good.

**Comment by lanej** [comment]
ah yes, the ole commit message format.

File: helm/action/list.go (hunk 1 above)
Comment (lanej) [inline helm/action/list.go]: just like to prevent accidental assignments instead of comparison.  it's a habit. i can back this out.

Comment (technosophos) [inline helm/action/list.go]: @carmstrong is right that the Go idiom is always to do `varname != nil`. Go will catch the case of accidental assignment.

**Comment by technosophos** [comment]
Other than switching back to the idiomatic `err != nil` in several places, I'm good with this.

Looks like if no argument is passed in, the `""` will be used.

**Comment by lanej** [comment]
@technosophos good to know. ill back those out.

**Comment by carmstrong** [comment]
@technosophos LGTY now?

**Comment by technosophos** [comment]
I am totally good with this now. Thank you very much @lanej!

**Pull request merged**